parser/wasm.rs

```diff
@@ -579,17 +654,52 @@ impl<'a> Parse<'a> for elements::CodeSection {
     }
 }
 
+// pos points behind the last position
+fn pop_stack(code: &[elements::Opcode], pos: &mut usize) -> Option<i32> {
```

Comment:
Woo! \o/ abstract interpretation!

I'd feel much more confident about the correctness of this method if we had a few simple `#[test]`s that covered each of the kinds of opcodes it can handle, a larger combination of opcodes that it can handle, and finally a test with opcodes it cannot handle.

Did you find that interpreting these opcodes was good enough for all intents and purposes to find global data accesses, or do more complicated sequences actually appear in practice that we don't yet handle with this function?

Reply:
Sadly the abstract interpretation doesn't actually work in practice, because locals are involved, which are much harder to determine.